bokeh/charts/_builder.py

```diff
@@ -76,8 +81,26 @@ class Builder(HasProps):
 
     x_range = Instance(Range)
     y_range = Instance(Range)
+    y_names = Seq(String)
+    x_names = Seq(String)
 
     palette = Seq(Color, default=DEFAULT_PALETTE)
+    source = Instance(ColumnDataSource)
+
+    index = Any(help="""
+    An index to be used for all data series as follows:
+
+    - A 1d iterable of any sort that will be used as
+       series common index
+
+    - As a string that corresponds to the key of the
+       mapping to be used as index (and not as data
+       series) if area.values is a mapping (like a dict,
+       an OrderedDict or a pandas DataFrame)
+
+   """)
```

Comment:
It might be worth getting @pzwang's opinion here too. There are actually some subtle differences, and the question would be how far do we go to reflect this taxonomy in our type systems. For example, a scatter plot doesn't really have/use an index. It is two columns plotted against one another, with no notion of "order". But a time series is actually a different kind of plot, one of the dimensions is an index, and is ordered. This distinction is lost at the level of models, but a high level charts interface may want to not ignore these details.